api/app/events/api.py

```diff
@@ -96,13 +99,86 @@ def get(self):
         return returnEvents, 200
 
 
+class EventAPI(EventMixin, restful.Resource):
+    @auth_required
+    def post(self):
+        args = self.req_parser.parse_args()
+
+        user_id = g.current_user["id"]
+        current_user = user_repository.get_by_id(user_id)
+        if not current_user.is_admin:
+            return FORBIDDEN
+
+        name = args['name']
+        description = args['description']
+        start_date = datetime.strptime(
+            (args['start_date']), '%Y-%m-%dT%H:%M:%S.%fZ')
+        end_date = datetime.strptime(
+            (args['end_date']), '%Y-%m-%dT%H:%M:%S.%fZ')
+
+        event = Event(name=name,
+                      description=description,
+                      start_date=start_date,
+                      end_date=end_date)
+
+        db.session.add(event)
+        db.session.commit()
+
+        try:
+
+            admin_event_role = EventRole('admin', user_id, event.id)
+            db.session.add(admin_event_role)
+            db.session.commit()
+        except Exception as e:
+            LOGGER.error(
+                'Failed to add event role for user id {} due to: {}'.format(user_id, e))
+            return errors.ADD_EVENT_ROLE_FAILED
```

Comment:
Similar reasons here. I think there's two places this code can be pushed down closer to it's domain objects. 

The first I think can be into a generic `add_role(event_id, role)` method in the AppUser model. It looks like the `event_roles` field is already there so you would create the `EventRole` object and append it to `current_user.event_roles`. A db.session.commit should then save it (I think, just test and double check)?

The second place you could do it would be doing something similar but on the Event object you created above (but you would modify it before you saved). It looks like you might still need to add the `event` field and `event_roles` field (use `AppUser` and `EventRoles` as an example). Again here you would append `EventRole` to the `event_roles` field. And then when you save to the db, the `Event` and `EventRole` should both appear with one call. 

Let me know what you think and if I made sense with my explanation.

Reply:
I will investigate this and get back to you.